src/tracing/registry.h

```diff
@@ -0,0 +1,77 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <tracing/backend.h>
+
+namespace chip {
+namespace Tracing {
+
+/// Registers a tracing backend to receive trace and logging data
+/// Until it is unregistered
+///
+/// All tracing backends MUST be unregistered before the application
+/// exits. Consider using [ScopedRegistration]
+///
+/// MUST be called with chip thread lock held (from chip main loop or
```

Comment:
```suggestion
/// MUST be called with Matter thread lock held (from Matter main loop or
```
and similar elsewhere?